instana/instrumentation/sanic_inst.py

```diff
@@ -0,0 +1,138 @@
+                    await wrapped(*args, **kwargs)
+                except Exception as exc:
+                    span.log_exception(exc)
+                    raise
+    else:
+        @wrapt.patch_function_wrapper('sanic.server', 'HttpProtocol.write_response')
+        def write_with_instana(wrapped, instance, args, kwargs):
+            response = args[0]
+            span = async_tracer.active_span
+            if span is None:
+                wrapped(*args, **kwargs)
+            else:
+                response_details(span=span, response=response)
+                try:
+                    wrapped(*args, **kwargs)
+                except Exception as exc:
+                    span.log_exception(exc)
+                    raise
+
+
+        @wrapt.patch_function_wrapper('sanic.server', 'HttpProtocol.stream_response')
+        async def stream_with_instana(wrapped, instance, args, kwargs):
+            response = args[0]
+            span = async_tracer.active_span
+            if span is None:
+                await wrapped(*args, **kwargs)
+            else:
+                response_details(span=span, response=response)
+                try:
+                    await wrapped(*args, **kwargs)
+                except Exception as exc:
+                    span.log_exception(exc)
+                    raise
+
+
+    @wrapt.patch_function_wrapper('sanic.app', 'Sanic.handle_request')
+    async def handle_request_with_instana(wrapped, instance, args, kwargs):
+
+        try:
+            request = args[0]
+            try:  # scheme attribute is calculated in the sanic handle_request method for v19, not yet present
+                if "http" not in request.scheme:
+                    return await wrapped(*args, **kwargs)
+            except AttributeError:
+                pass
+            headers = request.headers.copy()
+            ctx = async_tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
+            with async_tracer.start_active_span("asgi", child_of=ctx) as tracing_scope:
+                tracing_scope.span.set_tag('span.kind', 'entry')
+                tracing_scope.span.set_tag('http.path', request.path)
+                tracing_scope.span.set_tag('http.method', request.method)
+                tracing_scope.span.set_tag('http.host', request.host)
+
+                query = request.query_string
+
+                if isinstance(query, (str, bytes)) and len(query):
+                    if isinstance(query, bytes):
+                        query = query.decode('utf-8')
+                    scrubbed_params = strip_secrets_from_query(query, agent.options.secrets_matcher,
+                                                               agent.options.secrets_list)
+                    tracing_scope.span.set_tag("http.params", scrubbed_params)
+
+                if agent.options.extra_http_headers is not None:
+                    extract_custom_headers(tracing_scope, headers)
+                await wrapped(*args, **kwargs)
```

Comment:
Shall we also check the response and set `scope.span.mark_as_errored()`?

Reply:
At this point there is no access to the response, however the response is checked with one of the other patched methods, either `send` or `write_response` or `stream_response` depending on the sanic version. Additionally if there was 5xx status code the `exception_with_instana` method will add the error message if present to our span data. We can have a short look together in the flow as it might not be that clear.